pkg/operator/configobservation/images/observe_images.go

```diff
@@ -1,42 +1,163 @@
 package images
 
 import (
+	"bytes"
+	"encoding/json"
+
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
-	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
 	"github.com/openshift/library-go/pkg/operator/configobserver"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
 )
 
 func ObserveInternalRegistryHostname(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
 	listers := genericListers.(configobservation.Listers)
 	var errs []error
 	prevObservedConfig := map[string]interface{}{}
 
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
 	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
 	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
-		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		err := unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
 	}
 
 	if !listers.ImageConfigSynced() {
 		glog.Warning("images.config.openshift.io not synced")
 		return prevObservedConfig, errs
 	}
 
+	// now gather the cluster config and turn it into the observed config
 	observedConfig := map[string]interface{}{}
 	configImage, err := listers.ImageConfigLister.Get("cluster")
 	if errors.IsNotFound(err) {
 		glog.Warningf("image.config.openshift.io/cluster: not found")
-		return observedConfig, errs
+		return observedConfig, append(errs, err)
```

Comment:
this error shouldn't be an error that gets returned because it doesn't indicate a real failure.  there's just nothing to report

Reply:
will fix